geode-core/src/distributedTest/java/org/apache/geode/internal/cache/partitioned/PersistentPartitionHangsDuringRestartRegressionTest.java

```diff
@@ -0,0 +1,253 @@
+
+import java.io.Serializable;
+import java.util.Properties;
+import java.util.concurrent.CountDownLatch;
+
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+
+import org.apache.geode.CancelException;
+import org.apache.geode.cache.PartitionAttributesFactory;
+import org.apache.geode.cache.Region;
+import org.apache.geode.cache.RegionFactory;
+import org.apache.geode.cache.persistence.PartitionOfflineException;
+import org.apache.geode.distributed.internal.ClusterDistributionManager;
+import org.apache.geode.distributed.internal.DistributionMessage;
+import org.apache.geode.distributed.internal.DistributionMessageObserver;
+import org.apache.geode.internal.cache.InitialImageOperation;
+import org.apache.geode.internal.cache.InternalCache;
+import org.apache.geode.internal.cache.control.InternalResourceManager;
+import org.apache.geode.test.dunit.AsyncInvocation;
+import org.apache.geode.test.dunit.IgnoredException;
+import org.apache.geode.test.dunit.VM;
+import org.apache.geode.test.dunit.VMEventListener;
+import org.apache.geode.test.dunit.rules.CacheRule;
+import org.apache.geode.test.dunit.rules.DistributedDiskDirRule;
+import org.apache.geode.test.dunit.rules.DistributedRule;
+import org.apache.geode.test.junit.rules.serializable.SerializableTestName;
+
+/**
+ * Extracted from {@link PersistentPartitionedRegionRegressionTest}.
+ */
+@SuppressWarnings("serial,unused")
+public class PersistentPartitionHangsDuringRestartRegressionTest implements Serializable {
+
+  private static final long TIMEOUT_MILLIS = getTimeout().getValueInMS();
+
+  private static volatile CountDownLatch beforeBounceLatch;
+  private static volatile CountDownLatch afterBounceLatch;
+  private static volatile CountDownLatch waitToBounce;
+
+  private final transient VMEventListener vmEventListener = new VMEventListener() {
+    @Override
+    public void afterBounceVM(VM vm) {
+      afterBounceLatch.countDown();
+    }
+  };
+
+  private String partitionedRegionName;
+
+  private VM vmController;
+  private VM vm0;
+  private VM vm1;
+
+  @Rule
+  public DistributedRule distributedRule = new DistributedRule();
+
+  @Rule
+  public CacheRule cacheRule = new CacheRule();
+
+  @Rule
+  public SerializableTestName testName = new SerializableTestName();
+
+  @Rule
+  public DistributedDiskDirRule diskDirRule = new DistributedDiskDirRule();
+
+  @Before
+  public void setUp() {
+    beforeBounceLatch = new CountDownLatch(1);
+    afterBounceLatch = new CountDownLatch(1);
+
+    vmController = getController();
+    vm0 = getVM(0);
+    vm1 = getVM(1);
+
+    String uniqueName = getClass().getSimpleName() + "-" + testName.getMethodName();
+    partitionedRegionName = uniqueName + "-partitionedRegion";
+  }
+
+  @After
+  public void tearDown() {
+    removeVMEventListener(vmEventListener);
+    invokeInEveryVM(() -> InternalResourceManager.setResourceObserver(null));
+  }
+
+  /**
+   * RegressionTest for bug 42226. <br>
```

Comment:
I realize this was copy paste, but references to bug numbers we can't get to really shouldn't be propagated if we can help it.  The rest of the comment could still be helpful it seems

Reply:
On one hand, I do have access to those bugs, and when I work on code that references a TRAC ticket in comments, it is sometimes useful that I can go learn more about the history of it. The javadoc describes the bug well and includes the full summary.

On the other hand, 1/2 or more of the people working on Geode don't have access. The TRAC ticket is noise to them unless they ask to have someone pull up that ticket and share details of it-- maybe it ends up having value since some of us can grab and share that history with the Geode community?

I think I'm not the right person to remove these since I think they can be useful, but removing all TRAC tickets from comments/javadocs in Geode might be a good discussion on the dev-list. I do always convert test class and method names from testBug45666 to something more meaningful and then copy the full summary and try to describe what the bug was in javadocs. Sometimes I do delete the info if I'm sure there's no value in it. I wouldn't disapprove a PR that removes the references as I think both view points are valid.

Reply:
Sounds good, I'll look to start a dev thread.